diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb001f5..3039d75 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,13 +51,6 @@ include (CheckFunctionKeywords)
 include (CheckIncludeFiles)
 include (CheckTypeSize)
 
-# suppress format-truncation warning
-include (CheckCCompilerFlag)
-check_c_compiler_flag(-Wno-format-truncation HAS_NO_FORMAT_TRUNCATION)
-if (HAS_NO_FORMAT_TRUNCATION)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-format-truncation")
-endif()
-
 if (MSVC)
    # Turn off Microsofts "security" warnings.
    add_definitions( "/W3 /D_CRT_SECURE_NO_WARNINGS /wd4005 /wd4996 /nologo" )
@@ -636,8 +629,8 @@ include(CMakePackageConfigHelpers)
 
 write_basic_package_version_file(
   "${CMAKE_CURRENT_BINARY_DIR}/cmake/janssonConfigVersion.cmake"
-  VERSION ${JANSSON_VERSION}
-  COMPATIBILITY ExactVersion
+  VERSION ${JANSSON_DISPLAY_VERSION}
+  COMPATIBILITY SameMajorVersion
 )
 
 configure_package_config_file(
